### PR TITLE
[WFCORE-3360] Eliminate org.wildfly.extension.elytron.ProviderUtil

### PR DESCRIPTION
**Pull request by justinmcook**
Jira: https://issues.jboss.org/browse/WFCORE-3360

### CODE REVIEW

**Comment by darranl** [comment]
Can someone please add Justin to the list of approved contributors?

**Comment by bstansberry** [comment]
this is ok to test

File: elytron/src/main/java/org/wildfly/extension/elytron/HttpServerDefinitions.java [hunk 1]
```diff
@@ -197,8 +196,10 @@ static ResourceDefinition getProviderHttpServerMechanismFactoryDefinition() {
                     providerSupplier = Security::getProviders;
                 }
 
+                Predicate<Provider.Service> serviceFilter = (Provider.Service s) -> HttpServerAuthenticationMechanismFactory.class.getSimpleName().equals(s.getType());
+
                 return () -> {
-                    if ( ! isServiceTypeProvided(providerSupplier.get(), HttpServerAuthenticationMechanismFactory.class)) {
+                    if ( org.wildfly.security.util.ProviderUtil.findProviderService(providerSupplier, serviceFilter) == null ) {
```
[the remark this answers is not in the document]
Comment (justinmcook) [reply elytron/src/main/java/org/wildfly/extension/elytron/HttpServerDefinitions.java]: Fixed

File: elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreService.java [hunk 2]
```diff
@@ -151,7 +151,8 @@ public void start(StartContext startContext) throws StartException {
 
     private Provider resolveProvider() throws StartException {
         Provider[] candidates = providers.getOptionalValue();
-        Provider identified = identifyProvider(candidates == null ? Security.getProviders() : candidates, provider, KeyStore.class, type);
+        Supplier<Provider[]> providersSupplier = () -> candidates == null ? Security.getProviders() : candidates;
+        Provider identified = org.wildfly.security.util.ProviderUtil.findProvider(providersSupplier, provider, KeyStore.class, type);
```
[the remark this answers is not in the document]
Comment (justinmcook) [reply elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreService.java]: Fixed

**Pull request merged**